### ISSUE TITLE
pthread link errors on linux with GHC 8.2

### ISSUE THREAD

**Issue by Quintasan** [jaspervdj/hakyll#612]
Trying to install using `stack build --fast` with LTS Haskell 10.5 (GHC 8.2.2). Installation fails with pthread link errors:

```
    /tmp/stack32654/hakyll-4.10.0.0/rts/posix/OSThreads.c:137:0: error:
         error: undefined reference to 'pthread_create'

    /tmp/stack32654/hakyll-4.10.0.0/rts/posix/OSThreads.c:139:0: error:
         error: undefined reference to 'pthread_detach'

    /tmp/stack32654/hakyll-4.10.0.0/rts/posix/OSThreads.c:141:0: error:
         error: undefined reference to 'pthread_setname_np'

    /tmp/stack32654/hakyll-4.10.0.0/rts/posix/OSThreads.c:184:0: error:
         error: undefined reference to 'pthread_key_create'

    /tmp/stack32654/hakyll-4.10.0.0/rts/posix/OSThreads.c:192:0: error:
         error: undefined reference to 'pthread_getspecific'

    /tmp/stack32654/hakyll-4.10.0.0/rts/posix/OSThreads.c:203:0: error:
         error: undefined reference to 'pthread_setspecific'

    /tmp/stack32654/hakyll-4.10.0.0/rts/posix/OSThreads.c:212:0: error:
         error: undefined reference to 'pthread_key_delete'

    /tmp/stack32654/hakyll-4.10.0.0/rts/posix/OSThreads.c:371:0: error:
         error: undefined reference to 'pthread_kill'

    /tmp/stack32654/hakyll-4.10.0.0/includes/rts/OSThreads.h:59:0: error:
         error: undefined reference to 'pthread_mutex_trylock'
    collect2: error: ld returned 1 exit status
    `gcc' failed in phase `Linker'. (Exit code: 1)
```

Whole log: [build.log.txt](https://github.com/jaspervdj/hakyll/files/1720104/build.log.txt)

**Comment by jaspervdj** [comment]
This should be fixed in `hakyll-4.11.0.0`, see https://github.com/jaspervdj/hakyll/commit/480da307d22aff8ab3817d1586710c5f4ff6d779

**Comment by Quintasan** [comment]
@jaspervdj I see, thanks. Looks like LTS Haskell is out of date as usual and `allow-newer: true` seems to be ignored...
